spk/synocli-kernel/src/synocli-kernelmodule.sh

```diff
@@ -0,0 +1,347 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#########################################################################
+
+#------------------------------------------------
+# Make sure an argument was passed 
+#------------------------------------------------
+usage ()
+{
+echo
+printf '%10s %s\n' "Usage :" "$0 [-s|--spk <package>] [<insmod|rmmod|reload|status>] module1.ko module2.ko ..."
+printf '%20s %s\n' "Optional :" "[-c|--config <file>:<option>]"
+echo
+printf '%40s %s\n' "[-s|--spk <package>] : " "SynoCommunity package name containing kernel modules"
+printf '%40s %s\n' "[<insmod|rmmod|reload|status>] : " "Action to be performed"
+printf '%40s %s\n' "[-h|--help] : " "Print this help"
+printf '%40s %s\n' "[-v|--verbose] : " "Verbose mode"
+echo
+printf '%10s %s\n' "" "Examples :"
+printf '%20s %s\n' "" "$0 --spk synokernel-cdrom --verbose cdrom sr_mod status"
+printf '%20s %s\n' "" "$0 --spk synokernel-cdrom --config synokernel-cdrom.cfg:default status"
+echo 1>&2
+}
+
+#------------------------------------------------
+# Print detailed information for debugging
+#------------------------------------------------
+verbose ()
+{
+printf '%20s %s\n' "" "$0 (verbose)"
+printf '%60s %s\n' "SynoCommunity kernel driver package name (SPK)" "[${SPK}]"
+printf '%60s %s\n' "SynoCommunity configuration file (SPK_CFG)" "[${SPK_CFG}]"
+printf '%60s %s\n' "SynoCommunity configuration path (SPK_CFG_PATH)" "[${SPK_CFG_PATH}]"
+printf '%60s %s\n' "SynoCommunity configuration option (SPK_CFG_OPT)" "[${SPK_CFG_OPT}]"
+printf '%60s %s\n' "Synology NAS arch (ARCH)" "[${ARCH}]"
+printf '%60s %s\n' "Synology DSM version (DSM_VERSION)" "[${DSM_VERSION}]"
+printf '%60s %s\n' "Running kernel version (KVER)" "[${KVER}]"
+printf '%60s %s\n' "Module action insmod|rmmod|reload|status (ACTION)" "[${ACTION}]"
+printf '%60s %s\n' "Kernel modules path (MPATH)" "[${MPATH}]"
+printf '%60s %s\n' "Full kernel modules path (KPATH)" "[${KPATH}]"
+printf '%60s %s\n' "Device firmware path (FPATH)" "[${FPATH}]"
+printf '%60s %s\n' "Kernel objects list (KO_LIST)" "[${KO_LIST}]"
+printf '%60s %s\n' "Kernel objects found (KO_FOUND)" "[${KO_FOUND}]"
+}
+
+
+#------------------------------------------------
+# Modules can be passed as:
+#   <module>
+#   <module>.ko
+#   /<path>/<module>*
+# The following find the right module as needed
+#------------------------------------------------
+ko_path_match ()
+{
+   for ko in $KO_LIST
+   do
+      # first check if the name
+	  # matches to a path
+      if [ ! "$(echo $ko | grep '/')" ]; then
+         # Ensure to add .ko if needed
+         [ ! "$(echo $ko | grep '.ko$')" ] && ko=$ko.ko
+		 # Replace any '-' or '_' by '[-_]
+		 ko=$(echo ${ko} | sed -r 's/[-_]/[-_]/g')
+         # Find full module kernel object path
+         ko=$(find $KPATH -name $ko | awk -F'drivers/' '{print $2}')
```

Comment:
Why presume that the module will be in the drivers/ subdirectory?  Shouldn't this also support other kernel module paths (such as net/ etc? 
Line 136 also references it.

Reply:
Excellent point, I've updated the code to be include one additional directory level to the variable, which in turn leave's room for `net/` or any other.